tools/fuzzer/main.c

```diff
@@ -243,8 +244,7 @@ int fuzzer_send_msg(struct fuzz *fuzzer)
 	/* wait for ipc reply */
 	gettimeofday(&tp, NULL);
 	timeout.tv_sec  = tp.tv_sec;
-	timeout.tv_nsec = tp.tv_usec * 1000;
-	timeout.tv_nsec += 300000000; /* 300ms timeout */
```

Comment:
I think you should still keep the first of the two lines, otherwise you might unintentionally get a lower timeout.

Reply:
Facing the weirdest problem. When I set `timeout.tv_nsec` and then use `timeout.tv_sec += 1` or even 10s, it will still timeout. It is waiting for that many seconds but I'd still get a timeout. And if I compile the same thing without `timeout.tv_nsec` set it would work even with `timeout.tv_sec += 1`. Cannot figure out what possibly can be the problem here, we use the same thing in byt-host.c for boot timeout

this is from `byt-host.c`
```c
  timeout.tv_sec  = tp.tv_sec;
  timeout.tv_nsec = tp.tv_usec * 1000;
  timeout.tv_sec += 5;
```

and this is from `main.c`
```c
  timeout.tv_sec  = tp.tv_sec;
  timeout.tv_nsec = tp.tv_usec * 1000;
  timeout.tv_sec += 3; /* 3s timeout */
```

Any idea what could be the problem here?

Reply:
qemu? So you say that even with 10 seconds you get a timeout? That's odd I have to say...

Reply:
No, if I use 
```c
  timeout.tv_sec  = tp.tv_sec;
  timeout.tv_sec += 3; /* 3s timeout */
```
it works. However, when we set `tv_nsec` it won't work at any timeout
```c
  timeout.tv_sec  = tp.tv_sec;
  timeout.tv_nsec = tp.tv_usec * 1000;
  timeout.tv_sec += 3; /* 3s timeout */
```

Reply:
if tp.tv_usec is close to 1s, then adding something to it will wrap-around. You have to convert everything to ns, add the duration you want, then split it in s and ns again.

Reply:
@plbossart I don't think we add more than another 1 billion ns, so no actual wraparound happens; the number will still fit. But we could get an invalid value, for being 1 billion and some; in that case we do need to subtract 1 billion ns and add 1 second.

Reply:
@crazoes lets remove this change from our PR. Even the initial implementation is buggy, but this needs to be addressed in a separate patch.

Here how the current implementation looks:

```
    /* wait for ipc reply */
»       gettimeofday(&tp, NULL);
»       timeout.tv_sec  = tp.tv_sec;
»       timeout.tv_nsec = tp.tv_usec * 1000;
»       timeout.tv_nsec += 300000000; /* 300ms timeout */
```

But this is wrong because timeout.tv_nsec is 32 bits and if we want to add more than 4 seconds it will wrap around. So, sometimes the timeout will be in the past.